### PR TITLE
Add a command to open a stack in your browser

### PR DESCRIPTION
**Pull request by tomasmik**
How this works:

You can navigate to any repository on your local machine and execute `stack open` which will search for a stack which was created from a git repository you're currently in and has a project root set to your current subdir.

If more than 5 results are returned a very basic search function is invoked and you are able to either filter the returned results or still use arrow keys to select.

The returned list is never longer than 10 stacks and scrolling is automatically enabled if the list is longer than that. If only 1 stack is found, we don't display anything and instantly open a browser window.

Using params you can:
* Change the count of stacks returned (only 10 are shown anyway before scrolling is enforced)
* Ignore current subdir and return all stacks for the repository
* Take current branch in to account when searching for stacks

**Caveats**

* If users have a lot of stacks from the same repo+subdir they will have to adjust count parameter and filter. Though I suspect this won't be typical.
* `git` must be installed on the machine for this to work
* Not sure how windows defender would treat these `.Exec` commands. This should be fine, but you never know.

https://github.com/spacelift-io/spacectl/issues/54

### CODE REVIEW

**Review by Yantrio — APPROVED** [review]

File: internal/browser.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// OpenWebBrowser will try to open the default browser
+// on a the callers machine. It supports windows, darwin and windows.
+func OpenWebBrowser(url string) error {
```
Comment (cube2222) [inline internal/browser.go:13]: There's actually a nice little package we could use: https://github.com/pkg/browser

File: internal/cmd/stack/open_command.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	}
+
+	name, err := getRepositoryName()
+	if err != nil {
+		return err
+	}
+
+	return findAndSelect(cliCtx.Context, &stackSearchParams{
+		count:          count,
+		projectRoot:    subdir,
```
Comment (cube2222) [inline internal/cmd/stack/open_command.go:51]: Does this correctly interact with empty project roots?

**Review by cube2222 — APPROVED** [review]
with nits

File: internal/browser.go (hunk 1 above)
Comment (tomasmik) [reply internal/browser.go:13]: I would rather open a ticket and implement this in a separate ticket as it affects other parts of spacectl too 🤷🏻

File: internal/cmd/stack/open_command.go (hunk 2 above)
Comment (tomasmik) [reply internal/cmd/stack/open_command.go:51]: I have stacks: 
image

If I execute open in dir of a repo I get:
image

**Pull request merged**